src/test/java/com/iexec/sms/tee/config/TeeWorkerPipelineConfigurationTests.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2025 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.tee.config;
+
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.util.unit.DataSize;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+class TeeWorkerPipelineConfigurationTests {
+    private static final String VERSION = "v5";
+    private static final String PRE_IMAGE = "preImage";
+    private static final String PRE_FINGERPRINT = "preFingerprint";
+    private static final DataSize PRE_HEAP_SIZE = DataSize.parse("3GB");
+    private static final String PRE_ENTRYPOINT = "preEntrypoint";
+    private static final String POST_IMAGE = "postImage";
+    private static final String POST_FINGERPRINT = "postFingerprint";
+    private static final DataSize POST_HEAP_SIZE = DataSize.parse("3GB");
+    private static final String POST_ENTRYPOINT = "postEntrypoint";
+    private final TeeWorkerPipelineConfiguration.StageConfig preCompute = new TeeWorkerPipelineConfiguration.StageConfig(
+            PRE_IMAGE,
+            PRE_FINGERPRINT,
+            PRE_HEAP_SIZE,
+            PRE_ENTRYPOINT);
+    private final TeeWorkerPipelineConfiguration.StageConfig postCompute = new TeeWorkerPipelineConfiguration.StageConfig(
+            POST_IMAGE,
+            POST_FINGERPRINT,
+            POST_HEAP_SIZE,
+            POST_ENTRYPOINT);
+
+    private Validator validator;
+
+    @BeforeEach
+    void setUp() {
+        final ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
```

Comment:
My IDE shows me an advice here for the factory which implements `AutoCloseable`.
For such class, there is a try with-resource statement available in Java
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
The safe way to wirte this becomes:
```suggestion
        try (final ValidatorFactory factory = Validation.buildDefaultValidatorFactory()) {
            validator = factory.getValidator();
        }
```

I speak about this because we will see it again. It can appear as well when working with files.